src/utils/next.ts

```diff
@@ -141,11 +157,28 @@ export const scaffold =
       }
     }
 
-    // Update token data in session, in case it was refreshed
-    if (reqWithSession.session) {
-      reqWithSession.session.tokenData = ctx.z.getTokenData();
+    const orgId = ctx.query.orgId as string;
+
+    if (orgId) {
+      if (!ctx.user?.is_superuser) {
+        if (!hasOrg(reqWithSession, orgId)) {
+          try {
+            reqWithSession.session.organizations = await getOrganizations(ctx);
+          } catch (error) {
+            reqWithSession.session.organizations = null;
+          }
+          if (!hasOrg(reqWithSession, orgId)) {
+            return {
+              notFound: true,
+            };
+          }
+        }
+      }
```

Comment:
I'm sorry if this is annoying but I would love some comments here

Reply:
Very good suggestion!